.eslintrc.js

```diff
@@ -329,5 +330,34 @@ module.exports = {
         'google-camelcase/google-camelcase': 0,
       },
     },
+    {
+      'files': ['examples/**/*.js', '!examples/visual-tests/**/*.js'],
```

Comment:
I think I know what's going on with all those warnings. The negative glob in this override is being resolved to every file in the repo that doesn't match `examples/visual-tests/**/*.js`, instead of just those files under `examples/`.

Can we try using [`ignorePatterns`](https://eslint.org/docs/user-guide/configuring/ignoring-code#ignorepatterns-in-config-files) instead of the negative glob and see if the warnings go away?

Reply:
`ignorePatterns` is a top-level property, it can't be used on an override:

```
Error: ESLint configuration in .eslintrc.js is invalid:
	- Unexpected top-level property "overrides[3].ignorePatterns".
```

I'm not sure why this change triggers the warnings, but maybe we should handle them directly? Either `reportUnusedDisableDirectives: false` or actually remove the unused disable directives.

Reply:
My understanding is that they aren't unused directives. The negative glob in this PR is inadvertently disabling the rules for files outside `examples/`, which in turn is causing the directives to be flagged as unused.

Unless you think I've misunderstood the whole thing?

Reply:
You're absolutely right, my bad! This is fixed by giving `examples/` its own `.eslintrc.js`. PTAL